### PR TITLE
Updated resid to vice 3.3 r36112 to fix filters

### PR DESCRIPTION
**Pull request by AnaLogiC76**
Straight copy of commits 36112-36962 from VICE SVN.
As the filter-implementation was found to cause clipping and other
issues, it was declared "experimental" (copy filter.* to
filter8580new.*) and then reverted to r32576 with some minor changes on
top.

### CODE REVIEW

File: vice/src/vsync.c [hunk 1]
```diff
@@ -80,6 +80,13 @@ static int refresh_rate;
 /* "Warp mode".  If nonzero, attempt to run as fast as possible. */
 static int warp_mode_enabled;
 
+#ifdef __LIBRETRO__
```
Comment (rsn8887) [inline vice/src/vsync.c]: What is this? This has nothing to do with SID.

File: vice/src/vsync.c [hunk 2]
```diff
@@ -80,6 +80,13 @@ static int refresh_rate;
...
+int retro_warp_mode_enabled()
```
Comment (rsn8887) [inline vice/src/vsync.c]: This shouldn't be in this PR.

File: vice/src/vsync.h [hunk 3]
```diff
@@ -33,6 +33,10 @@
 #define VSYNC_DEBUG
 #endif
 
+#ifdef __LIBRETRO__
```
Comment (rsn8887) [inline vice/src/vsync.h]: This shouldn't be in this PR.

File: vice/src/resid/Makefile.am [hunk 4]
```diff
@@ -2,9 +2,19 @@
...
 AR = @AR@
 
+AM_CFLAGS = @VICE_CFLAGS@
+
+#AM_CXXFLAGS = @VICE_CXXFLAGS@
+
 noinst_LIBRARIES = libresid.a
 
-libresid_a_SOURCES = sid.cc voice.cc wave.cc envelope.cc filter.cc dac.cc extfilt.cc pot.cc version.cc
+if NEW_8580_FILTER
```
Comment (rsn8887) [inline vice/src/resid/Makefile.am]: Should this be defined or not? Remember Retroarch core is not using configure. We are using a makefile. Did you edit Makefile.libretro?

Comment (rsn8887) [reply vice/src/resid/Makefile.am]: None of your changes to Makefile.am etc. will make any difference. We are not using configure in Retroarch. We are using Makefile.libretro etc.

Comment (rsn8887) [reply vice/src/resid/Makefile.am]: Remember Retroarch core is not using configure. We are using Makefile.libretro and Makefile.common.

Did you test that no change to Makefile.common is needed for example?

File: vice/src/vsync.c (hunk 1 above)
Comment (AnaLogiC76) [reply vice/src/vsync.c]: You're correct, the changes in vsync.c/.h are part of a change to allow warp-mode with vsync enabled, should not be in here.
I'm new to git, what is the best way to fix this? Recreate branch, new commit, new pull request?

Comment (AnaLogiC76) [reply vice/src/vsync.c]: Regarding Makefile.am and configure.in, I applied all commits to resid even if they are not used to keep everything in this folder in sync with VICE trunk. I did not touch anything make-related and always build via Makefile.retroarch.
This also means theres currently no way to build for RA using filter8580new, but as they are broken and seem to be retired I saw no point in even trying to support building with them.

Comment (rsn8887) [reply vice/src/vsync.c]: There are many ways in git.

You could checkout the vsync files from the previous version manually and then commit them using the amend flag:
git checkout HEAD~1 vice/src/vsync.*
git add vice/src/vsync.*
git commit --amend
git push -f origin master 

The push -f is for force it is needed to update your remote branch cleanly.

**Comment by rsn8887** [comment]
Any progress on this? I can probably help and create a new PR. One that doesn't include the vsync stuff.

**Comment by rsn8887** [comment]
Ok, implemented in https://github.com/libretro/vice-libretro/commit/faf4315c3d17cc7d66531b31fe68e56eb2882a8e

**Comment by AnaLogiC76** [comment]
Thanks for fixing the commit and sorry for the late reply, I was away for an extended weekend.
What is the best way to discuss new features, e.g. warp-mode with vsync on, should I create a PR with what I have ready or create a topic on the libretro-forums?

**Comment by rsn8887** [comment]
I'd say make a PR. Is it already working on your branch? If not, maybe start the PR with "[WIP]" in the title.

**Pull request closed without merging**